### PR TITLE
Expose endpoint for getting competition admins

### PR DESCRIPTION
**Pull request by nilssonemil**
Further additions to the same resource will be the ability to post and
delete admins for competitions.

### CODE REVIEW

**Comment by nilssonemil** [comment]
> Why have a seperate GET endpoint for this and not include the information with the user? I know I’m thinking from a frontend perspective when I want it included in the user, wouldn’t it be easier not to have to do a separate api call. Or would that be mixing objects?

It could be included in both resources. The reason why I chose `/competitions/_/admins`, to at least start with, is that I think these resources would definitely be needed in time, and they are easier to make (fewer joins). Maybe at some point we could add admin information directly in the user, but I am not sure if it would make sense.

EDIT: You'd also have to make another call to the `/user/id` to make sure that the user is admin when making these calls. Then I think it semantically makes more sense to get the competitions admins and check if they match instead.

File: resources/public/openapi/spec.yml [hunk 1]
```diff
@@ -480,6 +503,22 @@ components:
       items:
         $ref: '#/components/schemas/Competition'
 
+    CompetitionAdmin:
```
Comment (hansenemily) [inline resources/public/openapi/spec.yml:539]: I think it might be me that's confused, but shouldn't CompetitionAdmin also contain a competition? Or is that unnecessary information? Because I guess that id it the id of the user? Or what id is it? Maybe add a description of the id as well?

File: src/amazestats/routes.clj [hunk 2]
```diff
@@ -71,7 +72,9 @@
            (GET "/:competition/divisions" [competition]
                 (find-divisions-by-competition competition))
            (GET "/:competition/teams" [competition]
-                (find-teams-by-competition competition)))
+                (find-teams-by-competition competition))
+           (GET "/:competition/admins" [competition]
+                (get-competition-admins competition)))
```
Comment (hansenemily) [inline src/amazestats/routes.clj:80]: Why is it get-competition-admins and not find-admins-by-competition? (Just out of curiosity)

**Review by hansenemily — CHANGES_REQUESTED** [review]

File: resources/public/openapi/spec.yml (hunk 1 above)
Comment (nilssonemil) [reply resources/public/openapi/spec.yml:539]: The ID belongs to the competition admin yes, and is from the start a user ID. However, it this point I do not see a reason to have the ID of the competition represented in the object as only the get for a specific ID of the competition exists. Therefore the resource you get already has a specific ID in the path of the request.

Depending on how the creation of admins are made, this object should probably be extended with a property called *competition*. If it should probably be a `writeOnly` property. However, I have no thought this through yet. At this point I do not see a reason to have a competition ID here.

File: src/amazestats/routes.clj (hunk 2 above)
Comment (nilssonemil) [reply src/amazestats/routes.clj:80]: Good question, I thought about this as well, but did not take action to change it because I did not really have that convincing arguments for/against it at the time.

Comment (hansenemily) [reply src/amazestats/routes.clj:80]: Okay, then let me rephrase: Why is it find-teams-by-competition and not get-competition-teams? Is it only that it sound better or is there a reason for it? To me it looks like they are doing the exact same thing?

File: resources/public/openapi/spec.yml (hunk 1 above)
Comment (hansenemily) [reply resources/public/openapi/spec.yml:539]: Write only property sound like a good idea and sure it can wait. 
I was just getting confused about what the id stood for, maybe a clarification is needed?

Comment (nilssonemil) [reply resources/public/openapi/spec.yml:539]: Some descriptions would probably be nice!

File: src/amazestats/routes.clj (hunk 2 above)
Comment (nilssonemil) [reply src/amazestats/routes.clj:80]: Honestly, I think I would prefix them all with `get` today. The reason (I think) behind the names are that I dabbled with allowing different query parameters (division, competition) to filter the result of a request for e.g. teams. Then we are asking the API to find teams based on some search criterion, whereas with a get I would expect the function call to be a bit more straight-forward.

Does that make sense? :D

Comment (hansenemily) [reply src/amazestats/routes.clj:80]: Yes, that makes sense. Thank you :)

**Review by hansenemily — APPROVED** [review]
